### ISSUE TITLE
ARG002: false negative case with `locals()`

### ISSUE THREAD

**Issue by Olegt0rr** [astral-sh/ruff#6576]
> ARG002 Unused method argument: `bar`

```python
def foo(bar: str):
    print(locals())
```

**Comment by charliermarsh** [comment]
I guess PyCharm doesn't consider these unused either, so we'll remove.

**Comment by dycw** [comment]
Hi @charliermarsh 

Is it possible to make this an option in that

- `locals()` considers `bar` as used (the current), or
- `locals()` does not consider `bar` as used

I am running into this as I often have an idiom

```
def foo(a, b, c):
    print_or_logging(locals())
    return a + b
```

But now I've missed the fact that `c` is in fact, unused.
